OpenRA.Game/Traits/Player/Shroud.cs

```diff
@@ -186,20 +186,22 @@ void Invalidate(IEnumerable<PPos> changed)
 				Hash += 1;
 		}
 
-		public static IEnumerable<PPos> ProjectedCellsInRange(Map map, WPos pos, WDist range, int maxHeightDelta = -1)
+		public static IEnumerable<PPos> ProjectedCellsInRange(Map map, WPos pos, WDist minRange, WDist maxRange, int maxHeightDelta = -1)
 		{
 			// Account for potential extra half-cell from odd-height terrain
-			var r = (range.Length + 1023 + 512) / 1024;
-			var limit = range.LengthSquared;
+			var r = (maxRange.Length + 1023 + 512) / 1024;
+			var minLimit = minRange.LengthSquared;
+			var maxLimit = maxRange.LengthSquared;
 
 			// Project actor position into the shroud plane
 			var projectedPos = pos - new WVec(0, pos.Z, pos.Z);
 			var projectedCell = map.CellContaining(projectedPos);
 			var projectedHeight = pos.Z / 512;
 
-			foreach (var c in map.FindTilesInCircle(projectedCell, r, true))
+			foreach (var c in map.FindTilesInAnnulus(projectedCell, minRange.Length / 1024, r, true))
 			{
-				if ((map.CenterOfCell(c) - projectedPos).HorizontalLengthSquared <= limit)
+				var dist = (map.CenterOfCell(c) - projectedPos).HorizontalLengthSquared;
+				if (dist <= maxLimit && (dist == 0 || dist > minLimit))
```

Comment:
Ideally, this would have been written as `if (dist < maxLimit && dist >= minLimit)`, but we have many actors with ranges on the cell boundary and this would have changed their vision footprints.

Reply:
Do we want to put a comment here explaining this?